mdal/frmts/mdal_ply.cpp

```diff
@@ -60,273 +69,236 @@ bool MDAL::DriverPly::canReadMesh( const std::string &uri )
   return true;
 }
 
+std::string MDAL::DriverPly::saveMeshOnFileSuffix() const
+{
+  return "ply";
+}
+
 std::unique_ptr<MDAL::Mesh> MDAL::DriverPly::load( const std::string &meshFile, const std::string & )
 {
   MDAL::Log::resetLastStatus();
+  Vertices vertices( 0 );
+  Faces faces( 0 );
+  Edges edges( 0 );
+  size_t maxSizeFace = 0;
 
-  std::ifstream in( meshFile, std::ifstream::in );
-  std::string line;
-
-  // Read header
   size_t vertexCount = 0;
   size_t faceCount = 0;
   size_t edgeCount = 0;
-  std::vector<std::string> chunks;
-  std::vector<MDAL::DriverPly::element> elements; // we will decant the element specifications into here
-  std::string proj = "";
-
-  if ( !std::getline( in, line ) )
-  {
-    MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " Header is corrupt" );
-    return nullptr;
-  }
 
-  /*
-  * The header is a format definition and a series of element definitions and/or comment lines
-  * cycle through these until end-header
-  */
-  do
+  //datastructures that will contain all of the datasets, categorised by vertex, face and edge datasets
+  std::vector<std::vector<double>> vertexDatasets; // contains the data
+  std::vector<std::pair<std::string, bool>> vProp2Ds; // contains the dataset name and a flag for scalr / vector
+  std::vector<std::vector<double>> faceDatasets;
+  std::vector<std::pair<std::string, bool>> fProp2Ds;
+  std::vector<std::vector<double>> edgeDatasets;
+  std::vector<std::pair<std::string, bool>> eProp2Ds;
+  std::unordered_map<std::string, std::pair<std::vector<double>, std::vector<int>>> listProps; // contains the list datasets as vector of values and vector indices for each face into the vector of values
+
+  libply::File file( meshFile );
+  if ( MDAL::Log::getLastStatus() != MDAL_Status::None ) { return nullptr; }
```

Comment:
you reseted the status at the beginning of the functions, it cannot be other than None?

Reply:
Actually no - since libply::File has been adapted to raise errors through MDAL_Status - this line will pick up any file load errors.